### PR TITLE
267 fix `@linksto` tag with `eval()` in last line of the evaluated code for `get_code`

### LINKED ISSUE #267: `get_code` looses comments, hence `# @linksto` tag if this is used in the last line with `eval()`
`eval` calls are not recognized as dependencies, even when explicitly tagged.

```
> data <- teal_data() |> eval_code(
+   code = '
+     expr <- quote(x <- x + 1)
+     x <- 0
+     eval(expr) # @linksto x
+   '
+ )
> get_code(data, datanames = "x") %>% cat
expr <- quote(x <- x + 1)
x <- 0
```

### PR DESCRIPTION
**Pull request by m7pr**
Close #267 

Used @gogonzo solution from this PR to fix this https://github.com/insightsengineering/teal.data/pull/262/files#diff-66a61facc4f5ca86215d63af40e2645b364950439051784916316ef78d5073c8R102-R110

### CODE REVIEW

**Comment by chlebowa** [comment]
```
data <- teal_data() |> eval_code(
  code = '
   expr <- quote(a <- x + 1)  # @linksto x
   x <- 0
   eval(expr) # @linksto x
 '
)
get_code(data, datanames = "x") %>% cat
```
That's better, thank you.

Is it possible to have `eval(expr)` be included automatically, like with `assign` and `data`?

**Comment by m7pr** [comment]
@chlebowa I'm not sure. We do extract `library()` automatically. For `data()` and `assign()` we only extended the detection so that it understands `data()` and `assign()` as ways of the assignment and they extract those lines if they refer to the object for which you are trying to get the code. For `eval({})` if the object is inside `{}` then this is also tackled. But I would not automatically extract `eval` always, cause it sometimes can refer to different object than the one you are asking the code for.

**Comment by chlebowa** [comment]
Oh, I don't mean always.
```
> data <- teal_data() |> eval_code(
+   code = '
+   expr <- quote(x <- x + 1)
+   x <- 0
+   eval(expr) 
+ '
+ )
> get_code(data, datanames = "x") %>% cat
expr <- quote(x <- x + 1)
x <- 0
```

We know to return the line that creates `expr` but we don't know to return the line where it is evaluated.

**Comment by m7pr** [comment]
I think at some point this could be possible but I would like to avoid trying this out before the very close release. We can try and start and issue to think about it. And currently let's merge this one that fixes an issue where `#@linksto x` tag is in the last line which also contains eval

**Comment by m7pr** [comment]
Created an issue for that in here https://github.com/insightsengineering/teal.data/issues/270

File: tests/testthat/test-get_code.R [hunk 1]
```diff
@@ -266,6 +266,19 @@ testthat::test_that("@linksto tag indicate affected object if object is assigned
 
 # @linksto ---------------------------------------------------------------------------------------------------------
 
+testthat::test_that("get_code does not break if @linksto is with eval in last line", {
```
Comment (chlebowa) [inline tests/testthat/test-get_code.R:269]: Does this cover only `eval` or other functions as well?

Comment (gogonzo) [reply tests/testthat/test-get_code.R:269]: Other as well. I put this solution first to fix `foo()` example from `get_code` docs

Comment (m7pr) [reply tests/testthat/test-get_code.R:269]: However we can rename the test to "get_code does not break if @linksto is in the last line"

Comment (m7pr) [reply tests/testthat/test-get_code.R:269]: Once this is merged to #263 I will do a thorough tests cleanup in there

File: R/utils-get_code_dependency.R [hunk 2]
```diff
@@ -130,7 +138,7 @@ fix_comments <- function(calls) {
       }
     }
   }
-  calls
+  Filter(function(x) nrow(x) != 0, calls)
```
Comment (gogonzo) [inline R/utils-get_code_dependency.R:141]:
```suggestion
  Filter(nrow, calls)
```

Comment (m7pr) [reply R/utils-get_code_dependency.R:141]: nice

File: tests/testthat/test-get_code.R [hunk 3]
```diff
@@ -267,6 +267,19 @@ testthat::test_that("detects function usage of the assignment operator", {
...
```
Comment (gogonzo) [inline tests/testthat/test-get_code.R:270]: Maybe clarification like this? 
```suggestion
testthat::test_that("get_code does not break if @linksto is put in the last line", {
  # In some cases R parses comment as a separate expression so the comment is not 
  # directly associated with this line of code. This situation occurs when `eval` is in the last
  # line of the code. Other cases are not known but are highly probable.
```

Comment (m7pr) [reply tests/testthat/test-get_code.R:270]: thanks @gogonzo

**Review by gogonzo — APPROVED** [review]

**Pull request merged**